### PR TITLE
Fix for #465  and  #72

### PR DESCRIPTION
**Pull request by drunkbear**
Hi,
I've been trying it for a few days and had no issues described in #465 and #72 so far.
Diff is a bit messy (sorry about that) - I've merged widget and tray window for simplicity's sake (shouldn't break anything). I can make another branch with actual fixes only if I'm wrong...

### CODE REVIEW

**Comment by flacjacket** [comment]
It looks like this introduces some problems with systray in the tests, in particular, it gives:
```
Traceback (most recent call last):
  File "/home/travis/build/qtile/qtile/test/utils.py", line 212, in runQtile
    state=self.state, no_spawn=self.no_spawn)
  File "/home/travis/build/qtile/qtile/libqtile/manager.py", line 161, in __init__
    self._process_screens()
  File "/home/travis/build/qtile/qtile/libqtile/manager.py", line 285, in _process_screens
    self._process_fake_screens()
  File "/home/travis/build/qtile/qtile/libqtile/manager.py", line 278, in _process_fake_screens
    s._configure(self, i, s.x, s.y, s.width, s.height, self.groups[i])
  File "/home/travis/build/qtile/qtile/libqtile/config.py", line 262, in _configure
    i._configure(qtile, self)
  File "/home/travis/build/qtile/qtile/libqtile/bar.py", line 196, in _configure
    i._configure(qtile, self)
  File "/home/travis/build/qtile/qtile/libqtile/widget/systray.py", line 126, in _configure
    self.screen = qtile.currentScreen.index
AttributeError: 'NoneType' object has no attribute 'index'
```

**Comment by drunkbear** [comment]
Thanks for a notice , I've fixed the formatting on that day and thought it would be just fine.
Current screen is wrong here, should've used bar screen instead.

**Comment by flacjacket** [comment]
Great, thanks, I'll try this branch out tonight.

File: libqtile/widget/systray.py [hunk 1]
```diff
@@ -190,13 +198,30 @@ def draw(self):
...
+                icon.unhide()
+                data = [
+                    self.qtile.conn.atoms["_XEMBED_EMBEDDED_NOTIFY"],
+                    xcffib.xproto.Time.CurrentTime,
+                    0,
+                    self.bar.window.window.wid,
+                    icon.protocol_version
+                ]
+                u = xcffib.xproto.ClientMessageData.synthetic(data, "I" * 5)
+                xcffib.xproto.ClientMessageEvent.synthetic(
```
Comment (flacjacket) [inline libqtile/widget/systray.py]: This event is never sent, you have to call `self.window.window.send_event()` on the event that `.synthetic()` return.

Comment (drunkbear) [inline libqtile/widget/systray.py]: Thanks, fixed that.

File: libqtile/widget/systray.py [hunk 2]
```diff
@@ -40,15 +40,16 @@
...
     _windowMask = EventMask.StructureNotify | \
+        EventMask.PropertyChange | \
         EventMask.Exposure
 
     def __init__(self, win, qtile, systray):
         window._Window.__init__(self, win, qtile)
         self.systray = systray
-        self.width = systray.icon_size
-        self.height = systray.icon_size
+        self.protocol_version = 1
```
Comment (flacjacket) [inline libqtile/widget/systray.py]: I'm just checking the XEmbed spec, should this be 0?

from http://standards.freedesktop.org/xembed-spec/xembed-spec-latest.html#lifecycle

Comment (drunkbear) [inline libqtile/widget/systray.py]: Yes, you're right. 
Moreover, the version sent on embedded notify should be the max version supported by embedder, not the one supported by the client that is being sent now.
I'll update it when I'm home.

**Comment by flacjacket** [comment]
Alright, this looks good, I'll push it in and hopefully it fixes the existing systray problems, thanks

**Comment by drunkbear** [comment]
Cool, thank you too! 
And I'm a bit late... Don't you mind adding xembed_info to property map 7b0e91fd452c83b991e510cac359528a36cef2d5? It will save some lines :)

**Comment by flacjacket** [comment]
Yeah, thanks for the update!

**Pull request merged**